doc/development.md

```diff
@@ -0,0 +1,107 @@
+    git@github.com:{user}/vdsm.git
+
+Enable oVirt packages for Fedora:
+
+    sudo dnf copr enable -y nsoffer/ioprocess-preview
+    sudo dnf copr enable -y nsoffer/ovirt-imageio-preview
+
+Enable
+[virt-preview](https://copr.fedorainfracloud.org/coprs/g/virtmaint-sig/virt-preview/)
+package to obtain latest qemu and libvirt versions (and update if needed):
+
+    sudo dnf copr enable @virtmaint-sig/virt-preview
+
+Install additional packages for Fedora, CentOS, and RHEL:
+
+    sudo dnf install -y `cat automation/check-patch.packages`
+
+Create virtual environment for vdsm:
+
+    python3 -m venv ~/.venv/vdsm
+    source ~/.venv/vdsm/bin/activate
+    pip install --upgrade pip
+    pip install -r docker/requirements.txt
+    deactivate
+
+## Running the tests
+
+Before running vdsm tests, activate the environment:
+
+    source ~/.venv/vdsm/bin/activate
+
+When done, you can deactivate the environment:
+
+    deactivate
+
+Create storage for the tests:
+
+    make storage
+
+Alternatively, storage tests can be completely setup and run by doing:
+
+    make tests-storage
+
+See the tests [README](../tests/README.md#running-the-tests)
+for more information.
+
+
+## Making new releases
+
+Release process of Vdsm version `VERSION` consists of the following
+steps:
+
+- Changing `Version:` field value in `vdsm.spec.in` to `VERSION`.
+
+- Updating `%changelog` line in `vdsm.spec.in` to the current date,
+  the committer, and `VERSION`.
+
+- Committing these changes, with subject "New release: `VERSION`" and
+  posting the patch to gerrit.
+
+- Verifying the patch by checking that the Jenkins build produced a
+  correct set of rpm's with the correct version.
+
+- Merging the patch (no review needed).
+
+- Tagging the commit immediately after merge with an annotated tag:
+  `git tag -a vVERSION`
+
+- Setting "Keep this build forever" for the check-merge Jenkins build.
+
+- Updating releng-tools with the new Vdsm version.  See releng-tools
+  repo (`git clone https://gerrit.ovirt.org/releng-tools`) and Vdsm
+  related patches there for examples.
+
+
+## CI
+
+Running tests locally is convenient, but before your changes can be
+merged, we need to test them on all supported distributions and
+architectures.
+
+When you submit patches to gerrit, oVirt's Jenkins CI will run its tests
+according to configuration in the stdci.yaml file.
+
+### Travis CI for storage patches
+
+oVirt's Jenkins CI is the integrated method for testing Vdsm patches,
+however for storage related patches we have to cover also 4k tests which
+are not covered currently by Jenkins CI. This can be achieved in a fast
+way manually and independently from gerrit by invoking Travis CI on your
+github branch:
+
+- Fork the project on github.
+- Visit https://travis-ci.org, register using your github account, and
+  enable builds for your Vdsm fork.
+- Push your changes to your github fork to trigger a build.
+
+See .travis.yml file for tested Travis platforms and tests configurations.
```

Comment:
We don't use Travis CI for a while, so we can drop this section. This is best done in a seaprate patch, before creating doc/developement.md